### DISCUSSION TITLE
Downgrade regolith-desktop to version 2.1 ?

### DISCUSSION THREAD

**Discussion by BlueDruddigon** [regolith-linux/regolith-desktop#792] [Q&A]
The ```regolith-desktop``` package has been updated to version 2.2 lately.
However, all my configurations have been broken after upgrading to 2.2.
I've tried downgrading the version of that package and its dependencies back to 2.1 and holding it there, but not yet.
What can I do now? Any solutions?

**Comment by kgilmer** [comment]
Let me see what I can do @BlueDruddigon..I'll reply back either way.

**Reply by kgilmer** [reply]
Ok so I've rebuilt 2.1 from source for Ubuntu Focal & Jammy and Debian Bullseye.  In order to take the downgrade, you'll need to change the URL that apt has for Regolith packages.  It depends on which version of Ubuntu you're using but essentially you swap "release" for "release-2_1".  Example:

Regolith 2.2 (from):

`/etc/apt/sources.list.d/regolith.list`:
```
deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg]  https://regolith-desktop.org/release-ubuntu-jammy-amd64 jammy main
```

Regolith 2.1 (to):

`/etc/apt/sources.list.d/regolith.list`:
```
deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] https://regolith-desktop.org/release_2-1-ubuntu-jammy-amd64 jammy main
```

Once you've updated `apt` with `sudo apt update` you'll then need to downgrade your regolith packages back to 2.1.  To do this I suggest uninstalling all regolith packages and then re-installing `regolith-desktop` and any other additional packages you had added.  Because you (just) updated the `apt` repo for Regolith for 2.1 this should cause that version to be installed.  Until/unless you change the `apt` URL again you will not receive any Regolith package updates.

Note: any user configs you have will be untouched by this process.  However if you modified Regolith system files (via `sudo`, eg `/usr/share/regolith/i3/...`), those will get overwritten by this process.
